syntax = "proto3";

package synapse;

message OpticalStimOptions {
  uint32 pixel_count = 1;
  repeated uint32 bit_width = 2;
  repeated uint32 sample_rate = 3;
  repeated uint32 gain = 4;
}

message OpticalStimConfig {
  uint32 peripheral_id = 1;
  repeated bool pixel_mask = 2;
  uint32 bit_width = 3;
  uint32 sample_rate = 4; // frame rate
  uint32 gain = 5;
}
